resources/design-document.md

```diff
@@ -0,0 +1,267 @@
+# Design Document
+
+## Creative Companion
+
+## 1. Problem Statement
+
+Develop a software program that populates a dynamic Word Pool by interfacing with an external thesaurus and rhyming dictionary API. This word pool should then allow users to seamlessly drag and drop these words into a target field, facilitating the creation of original song lyrics, poetry, or prose. The software should also support the organization, rearrangement, and editing of these text elements. The content can be saved as individual Projects and Word Pools to a database, ensuring that users can retrieve and further develop their work at any time.
+
+## 2. Top Questions to Resolve in Review
+
+1. Scope: How many features should we have?
+2. What would be the best way to save the text contents of the fields to the database - as JSON strings?
+3. What information is needed to create a personal project or word pool? What should the user be able to name it?
+4. How should project deletion work? Is there any data retention policy?
+5. What are the specifics of sorting projects or word pools by date created, date modified, or alphabetically?
+6. How should changes to a project be saved? Is there an autosave feature?
+7. Do word pools share similar features with projects like creation, deletion, sorting, and editing?
+8. What are the integration requirements of the APIs used to populate the word pools?
+
+## 3. Use Cases
+
+###### Project
+
+U1. As a CC customer, I want to be able to create a personal project
+
+U2. As a CC customer, I want to be able to delete a personal project
+
+U3. As a CC customer, I want to be able to change the name of a project
+
+U4. As a CC customer, I want to be able to view all my projects
+
+U5. As a CC customer, I want to be able to sort the projects by date created
+
+U6. As a CC customer, I want to be able to sort the projects by date last modified
+
+U7. As a CC customer, I want to be able to sort the projects by A-Z or Z-A
+
+U8. As a CC customer, I want to be able to open / import a selected project from the projects list to edit
+
+U9. As a CC customer, I want to be able to update / save changes to a project
+
+###### WordPool
+
+U10. As a CC customer, I want to be able to create a personal wordPool
+
+U11. As a CC customer, I want to be able to delete a personal wordPool
+
+U12. As a CC customer, I want to be able to change the name of a wordPool
+
+U13. As a CC customer, I want to be able to view all my wordPools
+
+U14. As a CC customer, I want to be able to sort the wordPools by date created
+
+U15. As a CC customer, I want to be able to sort the wordPools by date last modified
+
+U16. As a CC customer, I want to be able to sort the wordPools by A-Z or Z-A
+
+U17. As a CC customer, I want to be able to open / import a selected wordPool from the wordPool list to edit
+
+U18. As a CC customer, I want to be able to update / save changes to a wordPool
+
+###### API Queries
```

Comment:
I'm not sure how to interpret this sub-heading. The user stores below here seem like the core functionality of the app. What is the relationship to API queries?